provider/terraform/property_override.rb

```diff
@@ -44,6 +44,11 @@ module OverrideFields
       # Names of attributes that can't be set alongside this one
       attr_reader :conflicts_with
 
+      # Set to true in order to pass a reference to the full terraform resource
+      # into the flatten methods. This can be used to ignore api responses and
+      # override properties from what is set in the config.
+      attr_reader :full_context
```

Comment:
I'm ok with this, but wouldn't it be simpler to just always pass the ResourceData? We do that for expand right now even though it's not always used.

If you'd like to keep this though, I'd at least want the word "flatten" somewhere in the name since that's what it's actually for.

Reply:
It was primarily meant to keep the changes a bit more isolated and opt-in and I thought people would be opposed to changing all methods, but I'm totally happy to make the change global. I'll try and push it in it's own pr for cleanliness though.